### PR TITLE
Improve search relevance, fix #238

### LINKED ISSUE #238: Improve search pertinence
As a visitor I want to find popular and relevant datasets first.

### PR DESCRIPTION
**Pull request by davidbgk**
Not ideal for now because it requires an update of https://github.com/etalab/geozones to load all postal codes but better than nothing.

### CODE REVIEW

File: udata/core/dataset/search.py [hunk 1]
```diff
@@ -246,7 +249,8 @@ def serialize(cls, dataset):
                 geozones.append({
                     'id': zone.id,
                     'name': zone.name,
-                    'code': zone.code
+                    'keys': [key for key in zone.keys
```
Comment (noirbizarre) [inline udata/core/dataset/search.py]: I think there is now two path for this with etalab/geozones#4:
- zone.keys is a string
- zone.keys is an array of string

**Comment by noirbizarre** [comment]
I think spatial API, models,commands and/or search may need update to support keys as array.

File: udata/core/spatial/search.py [hunk 2]
```diff
@@ -28,7 +28,9 @@ def serialize(cls, zone):
         return {
             'zone_suggest': {
                 'input': list(set(
-                    [zone.name, zone.code] + zone.keys.values())),
+                    [zone.name, zone.code]
+                    + [key for key in zone.keys.values()]
```
Comment (noirbizarre) [inline udata/core/spatial/search.py]: Is it working for postal codes (when key is a list) ?

**Comment by noirbizarre** [comment]
r+

**Comment by davidbgk** [comment]
@noirbizarre did you try to run tests for that branch on your local instance?

**Comment by noirbizarre** [comment]
no, I will

**Pull request merged**